android/src/main/java/com/swmansion/rnscreens/Screen.kt

```diff
@@ -259,23 +346,143 @@ class Screen(context: ReactContext?) : FabricEnabledViewGroup(context) {
+                        -1 -> BottomSheetBehavior.STATE_HIDDEN
+                        0 -> BottomSheetBehavior.STATE_EXPANDED
+                        else -> throw IllegalArgumentException("Invalid detentCount/index combination $detentCount / $index")
+                    }
+                2 ->
+                    when (index) {
+                        -1 -> BottomSheetBehavior.STATE_HIDDEN
+                        0 -> BottomSheetBehavior.STATE_COLLAPSED
+                        1 -> BottomSheetBehavior.STATE_EXPANDED
+                        else -> throw IllegalArgumentException("Invalid detentCount/index combination $detentCount / $index")
+                    }
+                3 ->
+                    when (index) {
+                        -1 -> BottomSheetBehavior.STATE_HIDDEN
+                        0 -> BottomSheetBehavior.STATE_COLLAPSED
+                        1 -> BottomSheetBehavior.STATE_HALF_EXPANDED
+                        2 -> BottomSheetBehavior.STATE_EXPANDED
+                        else -> throw IllegalArgumentException("Invalid detentCount/index combination $detentCount / $index")
+                    }
+                else -> throw IllegalArgumentException("Invalid detentCount/index combination $detentCount / $index")
+            }
+
+        /**
+         * This method maps BottomSheetBehavior.State values to appropriate indices of detents array.
+         *
+         * @param state state of the bottom sheet
+         * @param detentCount length of array with detents fractions
+         *
+         * @throws IllegalArgumentException for invalid state / detentCount combinations
+         */
+        fun detentIndexFromSheetState(@State state: Int, detentCount: Int): Int {
+            return when (detentCount) {
+                1 -> when (state) {
+                    BottomSheetBehavior.STATE_HIDDEN -> -1
+                    BottomSheetBehavior.STATE_EXPANDED -> 0
+                    else -> throw IllegalArgumentException("Invalid state $state for detentCount $detentCount")
+                }
+                2 -> when (state) {
+                    BottomSheetBehavior.STATE_HIDDEN -> -1
+                    BottomSheetBehavior.STATE_COLLAPSED -> 0
+                    BottomSheetBehavior.STATE_EXPANDED -> 1
+                    else -> throw IllegalArgumentException("Invalid state $state for detentCount $detentCount")
+                }
+                3 -> when (state) {
+                    BottomSheetBehavior.STATE_HIDDEN -> -1
+                    BottomSheetBehavior.STATE_COLLAPSED -> 0
+                    BottomSheetBehavior.STATE_HALF_EXPANDED -> 1
+                    BottomSheetBehavior.STATE_EXPANDED -> 2
+                    else -> throw IllegalArgumentException("Invalid state $state for detentCount $detentCount")
+                }
+                else -> throw IllegalArgumentException("Invalid state $state for detentCount $detentCount")
+            }
+        }
```

Comment:
Yeah, I mean I can see it being placed is some kind of helper class. 

I placed it here, because it is the `Screen` that we attach the `BottomSheetBehavior` to, so it made sense for me to put this logic here, but I'll move it to helper.

Reply:
https://github.com/software-mansion/react-native-screens/pull/2045/commits/8ad9968bb83a15b6ee3f11e82a57b660f97ba624